nft_api/src/ingest/main.rs

```diff
@@ -289,38 +288,95 @@ pub async fn batch_init_service(
         .unwrap();
 }
 
+// Program parser function type.
+type ProgramParser = fn(Pubkey, i64, Vec<u8>, &Pool<Postgres>) -> BoxFuture<'_, ()>;
```

Comment:
Yes I see from your in progress branch how this would be a better prototype.

I think I am going to back this ProgramParser change out of this PR however, as I met with Austin and he has some potentially different ideas for program parser, and also then we can support getting your bubblegum changes in first as well.

Reply:
Put program parser idea on separate dev branch: https://github.com/jarry-xiao/candyland/tree/danenbm/program-parser-type-idea